alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/sharding/SolrShardingMetrics.java

```diff
@@ -48,18 +49,28 @@ public void updateMetrics() {
                 Tags shardTags = flocTags.and(Tags.of("shard", String.valueOf(shard.getInstance())));
                 Set<ShardInstance> shardInstances = solrShardingMetricsContainer.getShardInstances(shard);
                 int numberOfActiveShardInstances = (int) shardInstances.stream().filter(shardInstance ->
-                        solrShardingMetricsContainer.getReplicaState(shardInstance) == ReplicaState.ACTIVE).count();
+                        solrShardingMetricsContainer.getReplicaState(shardInstance).get() == ReplicaState.ACTIVE).count();
                 setAndCreateMetricIfNotExists("activeShardInstances", numberOfActiveShardInstances,
                         shardTags, "number");
                 shardInstances.forEach(shardInstance -> {
                     Tags instanceTags = shardTags.and(Tags.of("instanceHost", shardInstance.getHostName()));
-                    ShardState shardState = solrShardingMetricsContainer.getShardState(shardInstance);
+                    ShardState shardState = solrShardingMetricsContainer.getShardState(shardInstance).orElseGet(() -> {
+                        ShardState state = new ShardState();
+                        state = new ShardState();
+                        state.setLastIndexedChangeSetId(-1);
+                        state.setLastIndexedTxId(-1);
+                        state.setLastIndexedChangeSetCommitTime(-1);
+                        state.setLastIndexedTxCommitTime(-1);
+                        state.setMaster(false);
+                        state.setLastUpdated(-1);
+                        return state;
+                    });
                     setAndCreateMetricIfNotExists("lastIndexedChangeSetId",
                             shardState.getLastIndexedChangeSetId(), instanceTags, "number");
                     setAndCreateMetricIfNotExists("lastIndexedTxId", shardState.getLastIndexedTxId(),
                             instanceTags, "number");
                     setAndCreateMetricIfNotExists("instanceMode",
-                            solrShardingMetricsContainer.getReplicaState(shardInstance).ordinal(), instanceTags, "enumValue");
+                            solrShardingMetricsContainer.getReplicaState(shardInstance).get().ordinal(), instanceTags, "enumValue");
```

Comment:
Optional can be empty here ??

Reply:
Hmmm. I thought about that too but I assumed it wouldn't be null since no check was performed before.
Stupid to assume, I will add a check.